src/lib.rs

```diff
@@ -772,8 +772,23 @@ impl<T: fmt::Debug> fmt::Debug for Grid<T> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "[");
         if self.cols > 0 {
-            for (i, _) in self.data.iter().enumerate().step_by(self.cols) {
-                write!(f, "{:?}", &self.data[i..(i + self.cols)]);
+            if f.alternate() {
+                write!(f, "\n");
+                for (i, _) in self.data.iter().enumerate().step_by(self.cols) {
+                    let mut row = self.data[i..(i + self.cols)].iter().peekable();
+                    write!(f, "    [");
+                    while let Some(item) = row.next() {
+                        write!(f, " {item:width$.precision$?}", width = f.width().unwrap_or(self.data.iter().map(|i| format!("{i:?}").len()).max().unwrap()), precision = f.precision().unwrap_or(2));
```

Comment:
Hard to follow in one line. Line breaks would be great for improved readability

Reply:
Formatted, removed from loop, commented! Thanks for the feedback, makes a huge difference :)